### PR TITLE
Add presenters for edit and new pages

### PR DESCRIPTION
**Pull request by c-lliope**

### CODE REVIEW

File: spec/support/features/page_elements.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Features
+  def have_header(title)
```
Comment (c-lliope) [inline spec/support/features/page_elements.rb:2]: Using as a capybara matcher, ignoring

File: app/dashboards/customer_dashboard.rb [hunk 2]
```diff
@@ -13,6 +13,13 @@ def show_page_attributes
     attributes - [title_attribute]
   end
 
+  def edit_page_attributes
```
Comment (c-lliope) [inline app/dashboards/customer_dashboard.rb]: Rename to `form_attributes`

File: app/controllers/customers_controller.rb [hunk 3]
```diff
@@ -14,10 +15,11 @@ def show
   end
 
   def new
-    @customer = Customer.new
+    @presenter = FormPresenter.new(CustomerDashboard.new, Customer.new)
   end
 
   def edit
+    @presenter = FormPresenter.new(CustomerDashboard.new, @customer)
```
Comment (c-lliope) [inline app/controllers/customers_controller.rb:22]: These lines seem a bit repetitive

File: app/views/customers/_form.html.erb [hunk 4]
```diff
@@ -1,9 +1,10 @@
-<%= simple_form_for(@customer) do |f| %>
+<%= simple_form_for(@presenter.resource) do |f| %>
   <%= f.error_notification %>
 
   <div class="form-inputs">
-    <%= f.input :name %>
-    <%= f.input :email %>
+    <% @presenter.attribute_names.each do |attr_name| %>
```
Comment (graysonwright) [inline app/views/customers/_form.html.erb]: Hm... `attribute_names` is a different interface than the `ShowPresenter` uses...

File: lib/presenters/form_presenter.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+class FormPresenter < BasePresenter
+  def initialize(dashboard, resource)
+    @dashboard = dashboard
+    @resource = resource
+  end
+
+  attr_reader :resource
+
+  def attribute_names
```
Comment (c-lliope) [inline lib/presenters/form_presenter.rb:11]: Hm... `attribute_names` is a different interface than the `ShowPresenter` uses...
Not sure if that's worth changing in itself, though.

File: lib/presenters/form_presenter.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    dashboard.form_attributes
+  end
+
+  def page_title
+    resource.public_send(dashboard.title_attribute)
+  end
+
+  def index_path
+    route(nil, :customers)
```
Comment (c-lliope) [inline lib/presenters/form_presenter.rb]: this is a reference to `customers` - should replace it with a calculated value

File: lib/presenters/show_presenter.rb [hunk 7]
```diff
@@ -26,7 +26,9 @@ def edit_path
     route(:edit, resource_name, resource)
   end
 
+  attr_reader :resource
```
Comment (c-lliope) [inline lib/presenters/show_presenter.rb]: not sure why this needs to change...

File: spec/features/edit_page_spec.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  it "has forms for the customer's attributes" do
+    customer = create(:customer)
+
+    visit edit_customer_path(customer)
+
+    expect(page).to have_content("Name")
+    expect(page).to have_content("Email")
+  end
+
+  it "links to the edit page" do
```
Comment (c-lliope) [inline spec/features/edit_page_spec.rb]: This looks like copypasta

File: spec/lib/presenters/form_presenter_spec.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
+require "rails_helper"
+require "presenters/form_presenter"
+
+RSpec.describe FormPresenter do
+  describe "#page_title" do
+    it "is the resource's key attribute" do
```
Comment (c-lliope) [inline spec/lib/presenters/form_presenter_spec.rb]: `"is the value of the resource's key attribute"`

File: app/controllers/customers_controller.rb (hunk 3 above)
Comment (c-lliope) [inline app/controllers/customers_controller.rb:22]: "Duplication is far cheaper than the wrong abstraction"
– Sandi Metz

**Pull request merged**